### DISCUSSION TITLE
Enhancement: Toggle between rag and full context for attached documents

### DISCUSSION THREAD

**Discussion by Trapper4888** [danny-avila/LibreChat#2376] [Feature Requests & Suggestions]
### What features would you like to see added?

When using long context models, it would be nice to be able to pass pdf or txt files with the intent to include all of it into the context.

### More details

1. Add some rag/full toggle to the attached document(s). 
2. Furthermore, it would be pretty neat to have an option to include a document to a precise message OR to a whole conversation.
    - For the precise message it would be useful in cases I ask a question like "according to this user manual, what is the best..."
    - For the whole conversation, it would act kind of a system message (but I don't think it sould be included in the system message since some models don't support it) containing "memory".
      It would be even better to be able to include those in presets. This would allow me for example to have:
        -  A "Ask LibreChat documentation" preset, using Claude 3 Haiku and including "libre-chat-whole-doc.md" as a conversation-wide "full" document context. 
        - The same using Claude 3 Opus, including "libre-chat-whole-doc.md" as a conversation-wide "rag" document context. (which is the current functionality if I would add "libre-chat-whole-doc.md" to any message and enable "Resend Files")

### Which components are impacted by your request?

General, UI, Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Trapper4888** [comment]
"Furthermore, it would be pretty neat to have an option to include a document to a precise message OR to a whole conversation." this second section might deserve its own issue, since it would be interesting with the current default rag approach.

**Comment by danny-avila** [comment]
This is planned. On your second point, you toggle off “resend files” and then you can attach files on a message basis

**Comment by leversberg** [comment]
We have a RAG use case where we want to upload a CSV spreadsheet.
The current approach can only insert `k=4` chunks into the context, where one CSV data row equals one chunk. 
So LibreChat can only include 4 rows of a spreadsheet per message.

**Reply by danny-avila** [reply]
In my opinion the current RAG with a CSV is not optimal.

I’m currently looking for someone to help bring more attention to the RAG API because there are many improvements to be had and many things for me to maintain.

one of these improvements would include better handling for CSV across the board as well as reranking and configurable context chunks.

But it’s worth mentioning that The best solution for CSV is not “needle in a haystack” semantic search and rather a form of SQL agent over the tabular data, whether using literal SQL queries or some kind of data frame manipulation.

**Comment by richard-deetlefs** [comment]
Any update on this feature request?

**Comment by mzazon** [comment]
Curious if this has been addressed at all yet? I have a use case where I want to leverage 1M context of gemini pro 2.5 to pre-load a bunch of research documentation to synthesize. It is all in markdown already (I don't need to OCR it, so the mistral OCR functionality fails when I try to attach md files)

Simply just looking to attach text based files and include as part of context. Maybe I am doing something wrong!

**Comment by danny-avila** [comment]
Not yet addressed, it's coming to my attention and have a plan in place.
